test_echo_server.py

```diff
@@ -0,0 +1,26 @@
+import pytest
+from echo_client import echo_client
+
+
+def test_server_text():
+    """Test that a byte string works"""
+    text = "this is a test"
+    assert echo_client(text) == text
+
+
+def test_server_empty():
+    """Test that an empty string works"""
+    text = ""
+    assert echo_client(text) == text
+
+
+def test_server_unicode():
+    """Test that a unicode string works"""
+    text = u"this is a test"
+    assert echo_client(text) == text
```

Comment:
It would be far more meaningful to send a unicode string that contained non-ASCII characters.  That would cause `UnicodeDecodeError` or `UnicodeEncodeError` exceptions to raise for implicit byte/unicode conversions.  This proves nothing specific about the ability of your code to handle unicode strings because Python is implicitly converting them if needed.